### PR TITLE
feat: assert size of message and batch

### PR DESCRIPTION
**Pull request by 1abhishekpandey**
## Description

- Added the feature to ensure individual message size is not greater than 32KB.
- Added the feature to ensure batch size is not greater than 500KB.
- Added new API `deleteMessagesSync` to delete the message from the local DB/storage in a sync way.

**Fixes** # (*issue*)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] Version upgraded (project, README, gradle, podspec etc)
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: core/src/main/java/com/rudderstack/core/RudderUtils.kt [hunk 1]
```diff
@@ -54,4 +54,12 @@ object RudderUtils {
         }
     }
 
-}
\ No newline at end of file
+    internal fun getUTF8Length(message: String): Int {
```
Comment (itsdebs) [inline core/src/main/java/com/rudderstack/core/RudderUtils.kt:57]: make this an extension function String.utfLength()

File: core/src/test/java/com/rudderstack/core/AnalyticsTest.kt [hunk 2]
```diff
@@ -827,6 +857,23 @@ abstract class AnalyticsTest {
         verify(infraPlugin, times(1)).reset()
     }
 
+    private fun generateDataOfSize(msgSize: Int): String {
```
Comment (itsdebs) [inline core/src/test/java/com/rudderstack/core/AnalyticsTest.kt:860]: Can you explain this method? Also the name is not self explanatory

File: core/src/test/java/com/rudderstack/core/AnalyticsTest.kt [hunk 3]
```diff
@@ -827,6 +857,23 @@ abstract class AnalyticsTest {
...
+        return CharArray(msgSize).apply { fill('a') }.joinToString("")
+    }
+
+    private fun getNumberOfBatches(message: Message?, totalMessages: Int): Int {
```
Comment (itsdebs) [inline core/src/test/java/com/rudderstack/core/AnalyticsTest.kt:864]: Can you explain this method? Also the method and param names are not self explanatory.

File: core/src/main/java/com/rudderstack/core/RudderUtils.kt (hunk 1 above)
Comment (1abhishekpandey) [reply core/src/main/java/com/rudderstack/core/RudderUtils.kt]: Done.

File: core/src/test/java/com/rudderstack/core/AnalyticsTest.kt (hunk 3 above)
Comment (1abhishekpandey) [reply core/src/test/java/com/rudderstack/core/AnalyticsTest.kt]: I've improved it. Please review it again.

**Review by itsdebs — APPROVED** [review]

**Pull request merged**